gsdk/src/storage.rs

```diff
@@ -286,6 +286,26 @@ impl Api {
 
         Ok(mailbox)
     }
+
+    /// Get up to `count` messages from the mailbox
+    pub async fn all_mailbox(&self, count: u32) -> Result<Vec<(StoredMessage, Interval<u32>)>> {
+        let storage = self.storage().at(None).await?;
+        let query_key =
+            storage_address_root_bytes(&subxt::dynamic::storage_root("GearMessenger", "Mailbox"));
+
+        let keys = storage.fetch_keys(&query_key, count, None).await?;
+
+        let mut mailbox: Vec<(StoredMessage, Interval<u32>)> = vec![];
+        for key in keys.into_iter() {
```

Comment:
No need to convert `Vec` to `Iterator`.
```suggestion
        for key in keys {
```